packages/eas-cli/src/utils/expoCli.ts

```diff
@@ -1,8 +1,60 @@
 import spawnAsync from '@expo/spawn-async';
 import chalk from 'chalk';
+import { boolish } from 'getenv';
 import resolveFrom from 'resolve-from';
+import semver from 'semver';
 
 import Log from '../log';
+import { getExpoConfig } from '../project/expoConfig';
+import { memoize } from './expodash/memoize';
+
+// Aggressively returns `true` (UNVERSIONED, invalid SDK version format) to push users towards the versioned CLI.
+function gteSdkVersion(fromSdkVersion: string, sdkVersion: string): boolean {
+  if (fromSdkVersion === 'UNVERSIONED') {
+    return true;
+  }
+
+  try {
+    return semver.gte(fromSdkVersion, sdkVersion);
+  } catch {
+    return true;
+  }
+}
+
+/**
+ * @returns `true` if the project is SDK +46, has `@expo/cli`, and `EXPO_USE_LOCAL_CLI` is not set to a _false_ value.
+ */
+export function shouldUseVersionedExpoCLIExpensive(projectDir: string): boolean {
+  // Users can disable local CLI settings EXPO_USE_LOCAL_CLI=false
+  // https://github.com/expo/expo/blob/69eddda7bb1dbfab44258f468cf7f22984c1e44e/packages/expo/bin/cli.js#L10
+  // Run the environment variable check first as it's the cheapest.
+  const userDefinedVersionedCliEnabled = boolish('EXPO_USE_LOCAL_CLI', true);
+  if (!userDefinedVersionedCliEnabled) {
+    return false;
+  }
+
+  try {
+    // NOTE(EvanBacon): The CLI package could be available through a monorepo
+    // we need to ensure the project is specifically using a known supported Expo SDK version.
+    const { sdkVersion } = getExpoConfig(projectDir);
```

Comment:
My comment is still not addressed.
The ExpoConfig object should be passed from the top-level code (publish/index.ts):

```diff
diff --git a/packages/eas-cli/src/commands/update/index.ts b/packages/eas-cli/src/commands/update/index.ts
index abdde689..9ae47247 100644
--- a/packages/eas-cli/src/commands/update/index.ts
+++ b/packages/eas-cli/src/commands/update/index.ts
@@ -391,7 +391,7 @@ export default class UpdatePublish extends EasCommand {
       if (!skipBundler) {
         const bundleSpinner = ora().start('Building bundle...');
         try {
-          await buildBundlesAsync({ projectDir, inputDir });
+          await buildBundlesAsync({ exp, projectDir, inputDir });
           bundleSpinner.succeed('Built bundle!');
         } catch (e) {
           bundleSpinner.fail('Failed to build bundle!');
diff --git a/packages/eas-cli/src/project/publish.ts b/packages/eas-cli/src/project/publish.ts
index 7726e3b0..a12273b2 100644
--- a/packages/eas-cli/src/project/publish.ts
+++ b/packages/eas-cli/src/project/publish.ts
@@ -146,9 +146,11 @@ export async function buildUnsortedUpdateInfoGroupAsync(
 }
 
 export async function buildBundlesAsync({
+  exp,
   projectDir,
   inputDir,
 }: {
+  exp: ExpoConfig;
   projectDir: string;
   inputDir: string;
 }): Promise<void> {
@@ -157,7 +159,7 @@ export async function buildBundlesAsync({
     throw new Error('Could not locate package.json');
   }
 
-  if (shouldUseVersionedExpoCLI(projectDir)) {
+  if (shouldUseVersionedExpoCLI(projectDir, exp)) {
     await expoCommandAsync(projectDir, ['export', '--output-dir', inputDir, '--dump-sourcemap']);
   } else {
     // Legacy global Expo CLI
diff --git a/packages/eas-cli/src/utils/expoCli.ts b/packages/eas-cli/src/utils/expoCli.ts
index 4a2342eb..dc7a3912 100644
--- a/packages/eas-cli/src/utils/expoCli.ts
+++ b/packages/eas-cli/src/utils/expoCli.ts
@@ -1,3 +1,4 @@
+import { ExpoConfig } from '@expo/config-types';
 import spawnAsync from '@expo/spawn-async';
 import chalk from 'chalk';
 import { boolish } from 'getenv';
@@ -5,7 +6,6 @@ import resolveFrom from 'resolve-from';
 import semver from 'semver';
 
 import Log from '../log';
-import { getExpoConfig } from '../project/expoConfig';
 import { memoize } from './expodash/memoize';
 
 // Aggressively returns `true` (UNVERSIONED, invalid SDK version format) to push users towards the versioned CLI.
@@ -24,7 +24,7 @@ function gteSdkVersion(fromSdkVersion: string, sdkVersion: string): boolean {
 /**
  * @returns `true` if the project is SDK +46, has `@expo/cli`, and `EXPO_USE_LOCAL_CLI` is not set to a _false_ value.
  */
-export function shouldUseVersionedExpoCLIExpensive(projectDir: string): boolean {
+export function shouldUseVersionedExpoCLIExpensive(projectDir: string, exp: ExpoConfig): boolean {
   // Users can disable local CLI settings EXPO_USE_LOCAL_CLI=false
   // https://github.com/expo/expo/blob/69eddda7bb1dbfab44258f468cf7f22984c1e44e/packages/expo/bin/cli.js#L10
   // Run the environment variable check first as it's the cheapest.
@@ -36,7 +36,7 @@ export function shouldUseVersionedExpoCLIExpensive(projectDir: string): boolean
   try {
     // NOTE(EvanBacon): The CLI package could be available through a monorepo
     // we need to ensure the project is specifically using a known supported Expo SDK version.
-    const { sdkVersion } = getExpoConfig(projectDir);
+    const { sdkVersion } = exp;
 
     if (
       // If the version isn't defined then skip the check.
```